src/app/modules/webhook/__tests__/webhook.service.spec.ts

```diff
@@ -0,0 +1,409 @@
+      userId: MOCK_ADMIN_OBJ_ID,
+    })
+
+    jest.spyOn(Date, 'now').mockImplementation(() => MOCK_EPOCH)
+
+    // instantiate new form and save
+    const testEncryptedForm = new FormModel({
+      title: 'Test Form',
+      admin: preloaded.user._id,
+      responseMode: ResponseMode.Encrypt,
+      publicKey: 'fake-public-key',
+    })
+    await testEncryptedForm.save()
+
+    // initialise encrypted submussion
+    testEncryptedSubmission = new EncryptSubmissionModel({
+      form: testEncryptedForm._id,
+      authType: testEncryptedForm.authType,
+      myInfoFields: [],
+      encryptedContent: 'encrypted-content',
+      verifiedContent: 'verified-content',
+      version: 1,
+      webhookResponses: [],
+    })
+    await testEncryptedSubmission.save()
+
+    // initialise webhook related variables
+    testSubmissionWebhookView = testEncryptedSubmission.getWebhookView()
+
+    testSignature = formsgSdk.webhooks.generateSignature({
+      uri: MOCK_WEBHOOK_URL,
+      submissionId: testEncryptedSubmission._id,
+      formId: testEncryptedForm._id,
+      epoch: MOCK_EPOCH,
+    }) as string
+
+    testConfig = {
+      headers: {
+        'X-FormSG-Signature': `t=${MOCK_EPOCH},s=${testEncryptedSubmission._id},f=${testEncryptedForm._id},v1=${testSignature}`,
+      },
+      maxRedirects: 0,
+    }
+  })
+
+  describe('saveWebhookRecord', () => {
+    it('should return transform mongo error if database update for webhook fails', async () => {
+      // Arrange
+      const mockWebhookResponse = new Object({
+        ...MOCK_WEBHOOK_SUCCESS_RESPONSE,
+        signature: testSignature,
+        webhookUrl: MOCK_WEBHOOK_URL,
+      }) as IWebhookResponse
+
+      const mockDBError = new Error(DEFAULT_ERROR_MSG)
+
+      jest
+        .spyOn(EncryptSubmissionModel, 'addWebhookResponse')
+        .mockRejectedValueOnce(mockDBError)
+
+      // Act
+      const actual = await saveWebhookRecord(
+        testEncryptedSubmission._id,
+        mockWebhookResponse,
+      )
+
+      // Assert
+      const expectedError = transformMongoError(mockDBError)
+
+      expect(actual.isErr()).toBe(true)
+      expect(actual._unsafeUnwrapErr()).toEqual(expectedError)
+    })
+
+    it('should return submission not found error if submission id cannot be found in database', async () => {
+      // Arrange
+      const mockWebhookResponse = new Object({
+        ...MOCK_WEBHOOK_SUCCESS_RESPONSE,
+        signature: testSignature,
+        webhookUrl: MOCK_WEBHOOK_URL,
+      }) as IWebhookResponse
+
+      const invalidSubmission = new EncryptSubmissionModel()
+
+      // Act
+      const actual = await saveWebhookRecord(
+        invalidSubmission._id,
+        mockWebhookResponse,
+      )
+
+      // Assert
+      const expectedError = new SubmissionNotFoundError(
+        'Unable to find submission ID to update webhook response',
+      )
+
+      expect(actual.isErr()).toBe(true)
+      expect(actual._unsafeUnwrapErr()).toEqual(expectedError)
```

Comment:
you can mock the `addWebhookResponse` function to return a specific value, and check that the result returned matches that value.